Added file:// info in FAQ (#152)
leaningtech/labs: sites/cheerpx/src/content/docs/20-faq.md

`Uncaught CheerpX initialization failed: DataCloneError: DedicatedWorkerGlobalScope.postMessage: The SharedArrayBuffer object cannot be serialized. The Cross-Origin-Opener-Policy and Cross-Origin-Embedder-Policy HTTP headers can be used to enable this.`

This error occurs because CheerpX relies on [SharedArrayBuffer], which requires the site to be [cross-origin isolated]. To activate cross-origin isolation, ensure your site is served over HTTPS and include the following headers in your responses:

```
Cross-Origin-Embedder-Policy: require-corp
Cross-Origin-Opener-Policy: same-origin
```

By adding these headers to your server configuration you will enable cross-origin isolation and CheerpX will be able to start.

For more information on server configurations, please see our [server setup guide].

## Why Can't I Use the `file://` Protocol?

CheerpX requires certain browser capabilities that are not available when accessing files directly using the `file://` protocol. Here’s an explanation of the issue and the recommended solution:

### Security Restrictions and Cross-Origin Headers

> Modern browsers enforce strict security policies to protect users from potentially harmful actions. Accessing files through the `file://` protocol poses a security risk, as it could allow untrusted HTML files to interact with local resources. For instance, if you open a random HTML file received via email, it could potentially access sensitive files on your disk, leading to data breaches. Additionally, CheerpX needs to download multiple resources from its origin, including the WASM file, and relies on cross-origin isolation, which requires specific HTTP headers that can only be set when serving content over HTTP or HTTPS. Without these headers, the necessary security mechanisms cannot be enforced.

Reviewer: I would recommend to remove the HTML-via-email example. Makes the whole thing more difficult to read for no actual advantages. Also might make people worried that this is somehow connected to CheerpX

### Recommended Solution: Local Web Server Requirement

To properly utilize CheerpX, you must run a local web server. This allows you to access your HTML files with a URL that starts with `http://` or `https://`, such as `http://localhost:8080/`. Using a web server ensures that all necessary HTTP headers for cross-origin isolation are correctly applied.

For more information on server setups, please see our [server setup guide].

> [server setup guide]: :/docs/guides/nginx

Reviewer: This syntax for the link is not correct

`Uncaught CheerpX initialization failed: DataCloneError: DedicatedWorkerGlobalScope.postMessage: The SharedArrayBuffer object cannot be serialized. The Cross-Origin-Opener-Policy and Cross-Origin-Embedder-Policy HTTP headers can be used to enable this.`

This error occurs because CheerpX relies on [SharedArrayBuffer], which requires the site to be [cross-origin isolated]. To activate cross-origin isolation, ensure your site is served over HTTPS and include the following headers in your responses:

```
Cross-Origin-Embedder-Policy: require-corp
Cross-Origin-Opener-Policy: same-origin
```

By adding these headers to your server configuration you will enable cross-origin isolation and CheerpX will be able to start.

For more information on server configurations, please see our [server setup guide].

## Why Can't I Use the `file://` Protocol?

> CheerpX requires certain browser capabilities that are not available when accessing files directly using the `file://` protocol. Here’s an explanation of the issue and the recommended solution:

Reviewer: ```suggestion
CheerpX requires certain browser capabilities that are not available when accessing files directly using the `file://` protocol. Here’s an explanation of the issue and the recommended solution.
```

### Security Restrictions and Cross-Origin Headers

> Modern browsers enforce strict security policies to protect users from potentially harmful actions. Accessing files through the `file://` protocol poses a security risk, as it may allow untrusted HTML files to interact with local resources and cause to data breaches.

Reviewer: ```suggestion
Modern browsers enforce strict security policies to protect users from potentially harmful actions. Accessing files through the `file://` protocol poses a security risk, as it may allow untrusted HTML files to interact with local resources and cause data breaches.
```

> Modern browsers enforce strict security policies to protect users from potentially harmful actions. Accessing files through the `file://` protocol poses a security risk, as it may allow untrusted files to interact with local resources. Which may lead to data breaches.

Reviewer: ```suggestion
Modern browsers enforce strict security policies to protect users from potentially harmful actions. Accessing files through the `file://` protocol poses a security risk, as it may allow untrusted HTML files to interact with local resources and cause to data breaches.
```

> Modern browsers enforce strict security policies to protect users from potentially harmful actions. Accessing files through the `file://` protocol poses a security risk, as it can allow untrusted HTML files to interact with local resources. For example, opening random HTML files can access sensitive files on your disk, which may lead to data breaches.

Reviewer: ```suggestion
Modern browsers enforce strict security policies to protect users from potentially harmful actions. Accessing files through the `file://` protocol poses a security risk, as it can allow untrusted HTML files to interact with local resources. For example, opening random HTML files could access sensitive files on your disk, which may lead to data breaches.
```

Reviewer: Also remove _all_ mentioned of the HTML-via-email case. It's just confusing.

> Additionally, SharedArrayBuffer requires cross-origin isolation to function properly, which cannot be established when serving files with the `file://` protocol. This protocol restricts the ability to set crucial HTTP headers, such as `Cross-Origin-Opener-Policy` and `Cross-Origin-Embedder-Policy`.

Reviewer: ```suggestion
Additionally, [SharedArrayBuffer] requires cross-origin isolation to function properly, which cannot be established when serving files with the `file://` protocol. This protocol restricts the ability to set crucial HTTP headers, such as `Cross-Origin-Opener-Policy` and `Cross-Origin-Embedder-Policy`.
```

> Additionally, [SharedArrayBuffer] requires cross-origin isolation to function properly, which cannot be established when serving files with the `file://` protocol. This protocol restricts the ability to set crucial HTTP headers, such as `Cross-Origin-Opener-Policy` and `Cross-Origin-Embedder-Policy`.

Reviewer: ```suggestion
Additionally, [SharedArrayBuffer] requires cross-origin isolation to function properly, which cannot be established when serving files with the `file://` protocol. This protocol does not support setting the required HTTP headers: `Cross-Origin-Opener-Policy` and `Cross-Origin-Embedder-Policy`.
```

> Additionally, CheerpX needs to download multiple resources from its origin, including the WASM file, and relies on cross-origin isolation. This requires specific HTTP headers that can only be set when serving content over HTTP or HTTPS. Without these headers, the necessary security mechanisms cannot be enforced.

Reviewer: No, this is incorrect. SharedArrayBuffer depends on cross origin isolation. Downloading multiple resources is not allowed on file://

### Recommended Solution: Local Web Server Requirement

> To properly utilize CheerpX, you must run a local web server. This allows you to access your HTML files with a URL that starts with `http://` or `https://`, such as `http://localhost:8080/`. Using a web server ensures that all necessary HTTP headers for cross-origin isolation are correctly applied.

Reviewer: ```suggestion
To properly utilize CheerpX, you must run a local web server. This allows you to access your HTML files with a URL that starts with `http://` or `https://`, such as `http://localhost:8080/`. Using a web server ensures that all necessary HTTP headers for cross-origin isolation can be correctly applied.
```

> To properly utilize CheerpX, you must run a local web server. This allows you to access your HTML files with a URL that starts with `http://` or `https://`, such as `http://localhost:8080/`. Using a web server ensures that all necessary HTTP headers for cross-origin isolation can be correctly applied.

Reviewer: ```suggestion
To properly utilize CheerpX, you must run a local web server. This allows you to access your HTML files with a URL that starts with `http://` or `https://`, for example `http://localhost:8080/`. Using a web server ensures that all necessary HTTP headers for cross-origin isolation can be correctly applied.
```

> For more information on server setups, please see our [server setup guide].

Reviewer: ```suggestion
For more information on how to setup a HTTP server, please see our [server setup guide].
```

```

By adding these headers to your server configuration you will enable cross-origin isolation and CheerpX will be able to start.

## Why Can't I Use the `file://` Protocol?

CheerpX requires certain browser capabilities that are not available when accessing files directly using the `file://` protocol. Here’s an explanation of the issue and the recommended solution:

### Security Restrictions and Cross-Origin Headers

Modern browsers enforce strict security policies to protect users from potentially harmful actions. Accessing files through the `file://` poses a security risk, as it could allow untrusted HTML files to interact with local resources. For instance, if you open a random HTML file received via email, it could potentially access sensitive files on your disk, leading to data breaches.

> Additionally, CheerpX relies on cross-origin isolation, which requires specific HTTP headers that can only be set when serving content over HTTP or HTTPS. Without these headers, the necessary security mechanisms cannot be enforced. This is particularly important for CheerpX, as it needs to download multiple resources from its origin, including the WASM file.

Reviewer: The requirement of downloading multiple resource should be moved to the previous paragraph.

### Recommended Solution: Local Web Server Requirement

> To properly utilize CheerpX, you must run a local web server. This allows you to access your HTML files with a URL that starts with `http://` or `https://`, such as `http://localhost:8080/`. Using a web server ensures that all necessary security configurations and headers are correctly applied.

Reviewer: We should link back to the nginx configuration. Also there are no "security configurations" to apply.

```

By adding these headers to your server configuration you will enable cross-origin isolation and CheerpX will be able to start.

## Why Can't I Use the `file://` Protocol?

CheerpX requires certain browser capabilities that are not available when accessing files directly using the `file://` protocol. Here are the key reasons:

### 1. Security Restrictions

> Modern browsers enforce strict security policies that limit how local files can interact with web technologies. Accessing files through `file://` doesn’t allow the necessary cross-origin headers to be enabled, which are crucial for CheerpX to function properly.

Reviewer: It's not truly about cross origin, the idea is that you really don't want a random HTML file that arrived via email to access your private files on the disk. This is relevant since CheerpX needs to download multiple resources from its origin, in particular the WASM file.

> ### 2. Cross-Origin Headers

Reviewer: Consider collapsing these 2 points into a single paragraph. I am not convinced either way, so choose what seems to work better

CheerpX relies on cross-origin isolation, which requires specific HTTP headers. These headers can only be set when serving content over HTTP or HTTPS, not from the filesystem.

> ### 3. Local Web Server Requirement

Reviewer: This is not a "reason", it's how you avoid the problem
